coc/ext/triggers/cron.py

```diff
@@ -0,0 +1,215 @@
+    i.e. minute, hour, day of month, month, day of week with list, range and increment modifiers.
+    Name aliases for weekdays or months (e.g. Mon, Tue, ... and Jan, Feb, ...) are not supported
+
+    Attributes
+    ----------
+    cron_str: :class:`str`
+        the string representation of the Cron schedule
+
+    Examples
+    --------
+    "0 0 * * *": run every day at midnight
+    "15 0/4 1 * *": run at 15 minutes past every fourth hour on the first day of each month
+    "0 * 14,28 * *": run every hour on the 14th and 28th day of each month
+    "1/2 * * * 1-5": run every second hour starting at 1:00 AM on Monday through Friday
+    """
+
+    def __init__(self, cron_str: str):
+        try:
+            entries = CronParts(*cron_str.split())
+        except (TypeError, ValueError) as e:
+            raise CronParserError(
+                'Invalid Cron string. A Cron string must consist of exactly five entries separated by '
+                'whitespaces: minute, hour, day of month, month, day of week. Example: `0 0 * * *`'
+            ) from e
+
+        allowed_values = []
+        try:
+            for name, entry, limits in zip(NAMES, entries, LIMITS):
+                allowed_values.append(self.__parse_entry(name, entry, limits))
+        except ValueError as e:
+            raise CronParserError(
+                f'Invalid Cron string. {name.title()} is malformed. A Cron string element must either be the '
+                'wildcard `*` or contain one or more (comma-separated) entries. An entry must contain a time '
+                'indicator or a time range `start-end`, and it may contain an increment-suffix separated by a '
+                'slash `/`. Example: `4,12-18/2`'
+            ) from e
+
+        self.cron_str = cron_str
+        self._entries = entries
+        self.allowed_values = CronParts(*allowed_values)
+
+    def __str__(self) -> str:
+        return self.cron_str
+
+    def __eq__(self, other: Any):
+        # two Cron schedules can be considered equal if they allow the same values
+        # i.e. Cron('0 0 * * *') == Cron('0 0 1-31 * *')
+        return self.__class__ == other.__class__ and self.allowed_values == other.allowed_values
+
+    @staticmethod
+    def __parse_entry(name: str, entry: str, limits: List[int]) -> List[int]:
+        """Parse a single entry of the cron string"""
+
+        allowed_values = []
+        for part in entry.split(','):
+            # split off increment
+            if '/' in part:
+                part, increment = part.split('/')
+                increment = int(increment)
+                if increment <= 0:
+                    raise CronParserError('Invalid Cron string. Increments must be > 0')
+            else:
+                increment = None
+
+            if part == '*':  # wildcard
+                start, end = limits[0], limits[1]
+            elif '-' in part:  # time range
+                start, end = part.split('-')
```

Comment:
Should this raise a `CronParserError` if there are multiple minuses so the split spits out more than two items?

Reply:
Good point. Probably also should catch any value errors from people passing nonsense as well

Reply:
I think you already do that in line 52

Reply:
True, true. It's been a while since I touched this, forgot that I let that error propagate and handle it on the higher level. That will also catch the multiple minus case because that'll be a `ValueError`, too